### PR TITLE
Update transaction examples

### PR DESCRIPTION
**Pull request by sbmpost**

### CODE REVIEW

File: src/main/scala/chargepoint/docile/dsl/expectations/Ops.scala [hunk 1]
```diff
@@ -86,6 +86,7 @@ trait Ops[
     }
   }
 
+  // Ignoring unmatched messages only works if we force close the ocpp connection.
   def expectAllIgnoringUnmatched[T](expectations: IncomingMessageProcessor[T]*)(implicit awaitTimeout: AwaitTimeout): Seq[T] = {
```
Comment (sbmpost) [inline src/main/scala/chargepoint/docile/dsl/expectations/Ops.scala]: @reinierl I fixed the do-a-transaction example to handle the case of getting a ChangeConfigurationReq message before the StartTransactionRes. I am not sure if in OCPP this is even allows, but if it is, then we might consider a fix in docile-charger. The fix I would propose would be something along the lines of: 
```
  def sendSync[REQ <: CentralSystemReq, RES <: CentralSystemRes : ClassTag](
    req: REQ
  )(
    implicit
    reqRes: CentralSystemReqRes[REQ, RES],
    awaitTimeout: AwaitTimeout
  ): RES = {
    self.send(req)
    // self.expectIncoming(matching { case res: RES => res })
    self.expectAllIgnoringUnmatched(matching { case res: Res => res }).head
  }
```

File: src/main/scala/chargepoint/docile/dsl/expectations/Ops.scala [hunk 2]
```diff
@@ -86,6 +86,7 @@ trait Ops[
     }
   }
 
+  // Ignoring unmatched messages only works if we force close the ocpp connection.
```
Comment (sbmpost) [inline src/main/scala/chargepoint/docile/dsl/expectations/Ops.scala]: For more details, see the code comment above in OcppTest.scala

File: src/main/scala/chargepoint/docile/dsl/expectations/Ops.scala (hunk 1 above)
Comment (reinierl) [reply src/main/scala/chargepoint/docile/dsl/expectations/Ops.scala]: I would go a bit further and say that the proper fix would be to check the correlation of a response with a request. And then also to skip ahead in the queue of incoming messages to find a response for a given request.

I have heard from other people too about this problem of requests to the charge point messing up the charge point's expectations of responses to its requests. I hope I'll find some time in the coming weeks to come up with a solution.

**Comment by azolotko** [comment]
@sbmpost is this PR still relevant?

**Pull request closed without merging**